crates/holochain/src/sweettest/sweet_conductor_batch.rs

```diff
@@ -107,8 +107,13 @@ impl SweetConductorBatch {
 
     /// Let each conductor know about each others' agents so they can do networking
     pub async fn exchange_peer_info(&self) {
-        let envs = self.0.iter().map(|c| c.envs().p2p()).collect();
-        crate::conductor::p2p_store::exchange_peer_info(envs);
+        let mut all = Vec::new();
+        for c in self.0.iter() {
+            for env in c.envs().p2p().lock().values() {
+                all.push(env.clone());
+            }
+        }
+        crate::conductor::p2p_store::exchange_peer_info(all);
```

Comment:
Seems that this exchanges peer info across all spaces, could this be a bit too permissive, such that a test could pass which wouldn't pass in real life where the peer stores are segmented by space?

Reply:
Obviously the exchange isn't realistic--it'd be awesome if we could instantly know about everyone else : )

I don't think the space thing is a problem... or, rather, it'd actually be a problem if records from other spaces got mixed in a space store, but I'm pretty sure the logic inside that exchange function is delegating the records to the correct spaces and not mixing them.